### ISSUE TITLE
Oversized allocation warnings in the logs

### ISSUE THREAD

**Issue by denesb** [scylladb/scylla#3191]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20180206.6919c7434
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): GCE/Centos7

Saw a bunch of these in the journalctl logs:
```
febr 07 08:11:04 botond-benchmark-node3 scylla[4731]:  [shard 1] seastar_memory - oversized allocation: 1286144 bytes, please report: at   0x56c489
                                                        0x56d140
                                                        0x56eab4
                                                        0x56eb64
                                                        0xdaa76e
                                                        0xd77845
                                                        0xdb277c
                                                        0xd73f56
                                                        0xd77c52
                                                        0xd986d6
                                                        0x4b3716
                                                        0x4b39fb
                                                        0x504114
                                                        0x5085ce
                                                        0x56b68d
                                                        /lib64/libpthread.so.0+0x7e24
                                                        /lib64/libc.so.6+0xf834c
```
Resolved stack trace:
```
seastar::memory::cpu_pages::warn_large_allocation(unsigned long) at /usr/src/debug/scylla-666.development/seastar/core/memory.cc:629
seastar::memory::cpu_pages::check_large_allocation(unsigned long) at /usr/src/debug/scylla-666.development/seastar/core/memory.cc:637
 (inlined by) seastar::memory::cpu_pages::allocate_large(unsigned int) at /usr/src/debug/scylla-666.development/seastar/core/memory.cc:643
seastar::memory::allocate(unsigned long) at /usr/src/debug/scylla-666.development/seastar/core/memory.cc:1241
operator new(unsigned long) at /usr/src/debug/scylla-666.development/seastar/core/memory.cc:1644
__gnu_cxx::new_allocator<char>::allocate(unsigned long, void const*) at /opt/scylladb/include/c++/7/ext/new_allocator.h:111
 (inlined by) std::allocator_traits<std::allocator<char> >::allocate(std::allocator<char>&, unsigned long) at /opt/scylladb/include/c++/7/bits/alloc_traits.h:436
 (inlined by) std::_Vector_base<char, std::allocator<char> >::_M_allocate(unsigned long) at /opt/scylladb/include/c++/7/bits/stl_vector.h:172
 (inlined by) void std::vector<char, std::allocator<char> >::_M_range_insert<signed char const*>(__gnu_cxx::__normal_iterator<char*, std::vector<char, std::allocator<char> > >, signed char const*, signed char const*, std::forward_iterator_tag) at /opt/scylladb/include/c++/7/bits/vector.tcc:673
void std::vector<char, std::allocator<char> >::_M_insert_dispatch<signed char*>(__gnu_cxx::__normal_iterator<char*, std::vector<char, std::allocator<char> > >, signed char*, signed char*, std::__false_type) at /opt/scylladb/include/c++/7/bits/stl_vector.h:1406
 (inlined by) __gnu_cxx::__normal_iterator<char*, std::vector<char, std::allocator<char> > > std::vector<char, std::allocator<char> >::insert<signed char*, void>(__gnu_cxx::__normal_iterator<char const*, std::vector<char, std::allocator<char> > >, signed char*, signed char*) at /opt/scylladb/include/c++/7/bits/stl_vector.h:1132
 (inlined by) cql_transport::cql_server::response::write_value(std::experimental::fundamentals_v1::optional<seastar::basic_sstring<signed char, unsigned int, 31u> >) at /usr/src/debug/scylla-666.development/transport/server.cc:1795
cql_transport::cql_server::fmt_visitor::visit(cql_transport::messages::result_message::rows const&) at /usr/src/debug/scylla-666.development/transport/server.cc:1222 (discriminator 1)
cql_transport::cql_server::connection::make_result(short, seastar::shared_ptr<cql_transport::messages::result_message>, seastar::lw_shared_ptr<tracing::trace_state> const&, bool) at /usr/src/debug/scylla-666.development/transport/server.cc:1233
std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_M_erase(std::_Rb_tree_node<int>*) at /usr/src/debug/scylla-666.development/transport/server.cc:938
 (inlined by) apply at /usr/src/debug/scylla-666.development/seastar/core/apply.hh:36
 (inlined by) apply<cql_transport::cql_server::connection::process_execute(uint16_t, bytes_view, service::client_state)::<lambda(auto:87)>, seastar::shared_ptr<cql_transport::messages::result_message> > at /usr/src/debug/scylla-666.development/seastar/core/apply.hh:44
 (inlined by) apply<cql_transport::cql_server::connection::process_execute(uint16_t, bytes_view, service::client_state)::<lambda(auto:87)>, seastar::shared_ptr<cql_transport::messages::result_message> > at /usr/src/debug/scylla-666.development/seastar/core/future.hh:1276
std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_M_erase(std::_Rb_tree_node<int>*) at /usr/src/debug/scylla-666.development/seastar/core/future.hh:933
 (inlined by) run_and_dispose at /usr/src/debug/scylla-666.development/seastar/core/future.hh:413
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-666.development/seastar/core/reactor.cc:2495
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /usr/src/debug/scylla-666.development/seastar/core/reactor.cc:2893
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /opt/scylladb/include/c++/7/chrono:377
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-666.development/seastar/core/reactor.cc:3035
_Unwind_RaiseException at /usr/src/debug/scylla-666.development/seastar/core/reactor.cc:3983 (discriminator 3)
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-666.development/seastar/core/posix.cc:52
```
Looks like we use a vector somewhere where we shouldn't.

**Comment by denesb** [comment]
This was seen during a wide-partition read workload, for details see #3189

**Comment by slivne** [comment]
We are returning a page made of 10Kbyte rows - so a single result should be capped to be ~ 1MB, what is the default report size for the oversized allocation detector ?

**Comment by denesb** [comment]
I couldn't find the place in the code where the initial value is set.

**Comment by denesb** [comment]
I can see this warning every time I do reads for this profile.
Looks like we exploit and depend on the continuous storage of the `std::vector` so we cannot easily get rid of the large allocation by switching to `std:deque`.
If this is fine maybe we should temporary disable the large allocation warnings for this function.

**Comment by tgrabiec** [comment]
Fixed by a7c4d407cea1a5e2ebad35dccaa8bddd709ed07b
